src/evotorch/_distribute.py

```
@@ -0,0 +1,1221 @@
+        # We are being extra careful here for ensuring that `split_arguments` is a sequence of booleans.
+        # We want to actively prevent unexpected behavior that could be caused by these mistakes:
+        # - providing argument indices instead of a sequence of booleans
+        # - providing one or more argument names as strings, instead of a sequence of booleans
+        _actual_split_arguments = []
+        for split_arg in split_arguments:
+            if isinstance(split_arg, torch.Tensor) and (split_arg.ndim == 0):
+                _actual_split_arguments.append(bool(split_arg.to(device="cpu")))
+            elif isinstance(split_arg, (bool, np.bool_)):
+                _actual_split_arguments.append(bool(split_arg))
+            else:
+                raise TypeError("`split_arguments` was expected to contain booleans only")
```

Ruff (0.14.0)

Avoid specifying long messages outside the exception class

(TRY003)